functions-methods.js

```diff
@@ -2,35 +2,93 @@
+console.log(getEmailDomain("a.wiersma@outlook.com"));
+
+function getEmailDomain(emailAdres) {
+
+    let domeinnaam = [];
+
+    domeinnaam = emailAdres.split("@").reverse();
+
+    return domeinnaam[0];
+}
+
+/* Opdracht 2
+    [v] Schrijf een functie genaamd typeOfEmail, die een emailadres verwacht.
+    [v] De functie checkt of het emailadres een novi domein heeft (medewerker),
+    [v] een novi-education domein (student),
+    [v] of extern domein (zoals gmail of outlook)
+        ---- Verwachte uitkomsten:
+    [v] typeOfEmail("n.eeken@novi-education.nl") geeft "Student"
+    [v] typeOfEmail("t.mellink@novi.nl") geeft geeft "Medewerker"
+    [v] typeOfEmail("novi.nlaapjesk@outlook.com") geeft geeft "Extern" <-- deze moet het ook doen!
+    [v] typeOfEmail("a.wiersma@outlook.com") geeft "Extern"
+*/
+
+console.log("================opdracht2======================");
+console.log(typeOfEmail("n.eeken@novi-education.nl"));
+console.log(typeOfEmail("t.mellink@novi.nl"));
+console.log(typeOfEmail("novi.nlaapjesk@outlook.com"));
+console.log(typeOfEmail("a.wiersma@outlook.com"));
+
+function typeOfEmail(emailAdres) {
+
+    emailAdres = getEmailDomain(emailAdres)
+    let answer = "Extern";
+
+    if (emailAdres.includes("novi")) {
+        answer = "Medewerker";
+    }
+    if (emailAdres.includes("novi-education")) {
+        answer = "Student";
+    }
+    return answer;
+}
+
+/* Opdracht 3
+    [v] Schrijf een functie genaamd checkEmailValidity,
+    [V] die een emailadres verwacht en checkt of het emailadres valide is.
+    [v] De functie returned true of false, afhankelijk van de uitkomst.
+    [v] Een emailadres is valide wanneer:
+    [v] * Er een @ in voorkomt
+    [v] * Er géén , in voorkomt
+    [v] * Er géén . in voorkomt als allerlaatste karakter (dus hotmail.com is valide, net als outlook.nl, maar outlooknl. niet)
+        ---- Verwachte uitkomsten:
+    [v] checkEmailValidity("n.eeken@novi.nl") geeft true - want @ en punt op de juiste plek
+    [v] checkEmailValidity("tessmellink@novi.nl") geeft true - want @ en punt op de juiste plek
+    [v] checkEmailValidity("n.eekenanovi.nl") geeft false - want geen @
+    [v] checkEmailValidity("n.eeken@novinl.") geeft false - want de punt mag niet als laatst
+    [v] checkEmailValidity("tessmellink@novi,nl") geeft false - want er staat een komma in
+ */
+
+console.log("================opdracht3======================");
+console.log(checkEmailValidity("n.eeken@novi.nl"));
+console.log(checkEmailValidity("tessmellink@novi.nl"));
+console.log(checkEmailValidity("n.eekenanovi.nl"));
+console.log(checkEmailValidity("n.eeken@novinl."));
+console.log(checkEmailValidity("tessmellink@novi,nl"));
+
+function checkEmailValidity(emailAdres) {
+
+    let answer = false;
+    let check = [];
+    check = emailAdres.split("").reverse();
+
+    if (emailAdres.includes("@") && !emailAdres.includes(",") && check.indexOf(".", 0)) {
+        answer = true;
+    }
+    return answer;
```

Comment:
Je zou de uitkomst van de expressie meteen terug kunnen geven:
```suggestion
    return emailAdres.includes("@") && !emailAdres.includes(",") && check.indexOf(".", 0);
```